src/app/users/users-sign-in/users-sign-in.component.ts

```diff
@@ -110,8 +131,33 @@ export class UsersSignInComponent implements OnDestroy, OnInit {
     }
   }
 
+  recoverPassword(data) {
+    this.showResetPasswordFormErrors = true;
+    if (this.recoverPasswordForm.valid) {
+      this.store.dispatch(new RecoverPassword(data));
+      this.resetPasswordForm.get('username').setValue(data.username);
+      this.isRecoverFormSubmitted = true;
+      this.showResetPasswordFormErrors = false;
+    }
+  }
+
   resetPassword(data) {
-    this.resetModalRef.close();
+    this.showResetPasswordFormErrors = true;
+    if (this.resetPasswordForm.valid) {
+      this.openPgpService.generateUserKeys(data.username, data.password)
+        .then(keysData => {
+          const requestData = {
+            code: data.code,
+            username: data.username,
+            password: data.password,
+            private_key: keysData.private_key,
+            public_key: keysData.public_key,
```

Comment:
it can be written like this ::
```
 const requestData = {
           ...keysData,
           ...data
          };

```

Reply:
`data` contains `confirmPwd` field but I have updated the `...keysData` part.